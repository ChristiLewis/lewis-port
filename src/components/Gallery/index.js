//IMPORT ANY IMAGES- ADD THE SRC ATTRIBUTE OF {COVERIMAGE}
import coverImage from "../../assets/cover/0.jpg";
//MAKING A NEW REACT COMPONENT FOR THE GALLERY SECTION 1ST IMPORT 2ND FUNCTIONS 3RD EXPORT
import React from 'react';
import { capitalizeFirstLetter } from '../../utils/helpers';
import photo from "../../assets/physical/business-finance/0.jpg";


function Gallery(props) {
    const currentCategory = {
        name: "business-finance",
        description:
          "Digital applications and physical business and finance projects",
      };
    return (
        <section className="my-5">
            <h1>{capitalizeFirstLetter(currentCategory.name)}</h1>
            <img src={coverImage} className="my-2" style={{ width: "100%" }} alt="cover" />
            <p>{currentCategory.name}</p>
            <div className="flex-row">
                <img
                    src={photo}
                    alt="Business and Finance Examples"
                    className="img-thumbnail mx-1"
                />
            </div>
        </section>

    );
}

export default Gallery;