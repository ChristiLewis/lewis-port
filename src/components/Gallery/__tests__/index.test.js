//THIS FILE GOES INTO EACH SRC/COMPONENTS/COMPONENTNAME-PASCALCASE/__TESTS__/INDEX.TEST.JS-THIS ONE IS FOR THE COMPONENT SECTION
import React from 'react';

//IMPORT THE REACT TESTING LIBRARY
import { render, cleanup } from '@testing-library/react';

//IMPORT THE JEST-DOM PACKAGE
import '@testing-library/jest-dom/extend-expect';

//IMPORT THE COMPONENT
import Gallery from '..';

const business = { name: "business-finance", description: "Digital asset tracker app; physical offices, a bank, and a medical project" };

//CONFIGURE THE TEST ENV BY CALLING CLEANUP() VIA AFTEREACH GLOBAL FUNCTION FROM JEST
afterEach(cleanup);

//DECLARE THE COMPONENT BEING TESTED VIA DESCRIBE()
describe('Gallery is rendering', () => {
    //RENDERS COMPONENT TEST
    it('renders', () => {
        render(<Gallery currentCategory={business} />);
    });

    //CREATE A TEST CASE TO COMPARE SNAPSHOTS OF THE DOM NODE STRUCTURE- HERE INSIDE THE DESCRIBE CALLBACK FUNCTION BODY- BENEATH THE RENDER TEST
    it('matches snapshot DOM node stucture', () => {
        //RETURN THE SNAPSHOT
        const { asFragment } = render(<Gallery currentCategory={business} />);
        //COMPARE - CONTRAST W/ GOAL TO MATCH
        expect(asFragment()).toMatchSnapshot();
    })

    //TEST TO CHECK THE TITLE
    it('renders', () => {
        const { getByTestId } = render(<Gallery currentCategory={business} />)
        expect(getByTestId('h1tag')).toHaveTextContent('Business-finance')
    })

})
