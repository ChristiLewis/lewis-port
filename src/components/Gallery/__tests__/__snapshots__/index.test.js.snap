// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery is rendering matches snapshot DOM node stucture 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="h1tag"
    >
      Business-finance
    </h1>
    <p>
      Digital asset tracker app; physical offices, a bank, and a medical project
    </p>
    <div>
      <div
        class="flex-row"
      >
        <img
          alt="Asset graph"
          class="img-thumbnail mx-1"
          src="0.jpg"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
